src/platform/plugins/shared/dashboard/server/dashboard_saved_object/dashboard_saved_object.ts

```diff
@@ -79,6 +79,14 @@ export const createDashboardSavedObjectType = ({
       },
       optionsJSON: { type: 'text', index: false },
       panelsJSON: { type: 'text', index: false },
+      sections: {
+        properties: {
+          id: { type: 'keyword', index: false, doc_values: false },
+          title: { type: 'keyword', index: false, doc_values: false },
+          order: { type: 'integer', index: false, doc_values: false },
+          collapsed: { type: 'boolean', index: false, doc_values: false },
+        },
+      },
```

Comment:
Given that these are all `index: false` we should avoid creating new mappings for `sections*`, unless there is a reason other than convention for including them here?

Reply:
When I remove `sections` and its properties, I get the following error when trying to save a dashboard with a section:

```
Content management client error: [[1:204] mapping set to strict, dynamic introduction of [sections] within [dashboard] is not allowed: strict_dynamic_mapping_exception
	Root causes:
		strict_dynamic_mapping_exception: [1:204] mapping set to strict, dynamic introduction of [sections] within [dashboard] is not allowed]: [1:204] mapping set to strict, dynamic introduction of [sections] within [dashboard] is not allowed
```

Am I doing something wrong here? 🙈

Reply:
When I remove `sections` and its properties, I get the following error when trying to save a dashboard with a section:

```
Content management client error: [[1:204] mapping set to strict, dynamic introduction of [sections] within [dashboard] is not allowed: strict_dynamic_mapping_exception
	Root causes:
		strict_dynamic_mapping_exception: [1:204] mapping set to strict, dynamic introduction of [sections] within [dashboard] is not allowed]: [1:204] mapping set to strict, dynamic introduction of [sections] within [dashboard] is not allowed
```

When I try to keep `sections` around and just define it as `type: 'object'` without explicitly defining the properties, it just pushes the error down one level:

```
Content management client error: [[1:211] mapping set to strict, dynamic introduction of [id] within [dashboard.sections] is not allowed: strict_dynamic_mapping_exception
	Root causes:
		strict_dynamic_mapping_exception: [1:211] mapping set to strict, dynamic introduction of [id] within [dashboard.sections] is not allowed]: [1:211] mapping set to strict, dynamic introduction of [id] within [dashboard.sections] is not allowed
```

Am I doing something wrong here? 🙈

Reply:
I think, @jloleysens please correct me if I'm wrong, we can set the mapping like this:
```suggestion
      sections: {
        properties: {},
        dynamic: false
      },
```

I believe [setting `dynamic: false`](https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic.html) will allow the fields to exist in `_source`, but we won't have mappings for those fields.

Reply:
That seems to have worked!! 824ea407b5ad8b2e4ed0d2eb0de48fde711aca02

Thanks Nick 🙇